packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -711,6 +711,32 @@ export async function deleteExperimentByIdForOrganization(
   }
 }
 
+/**
+ * Delete experiments where the provided project is the only project for that experiment
+ * @param projectId
+ * @param organization
+ * @param user
+ */
+export async function deleteAllExperimentsForAProject({
+  projectId,
+  organization,
+  user,
+}: {
+  projectId: string;
+  organization: OrganizationInterface;
+  user: EventAuditUser;
+}) {
+  const experimentsToDelete = await ExperimentModel.find({
+    organization: organization.id,
+    project: projectId,
+  });
+
+  for (const experiment of experimentsToDelete) {
+    await experiment.delete();
```

Comment:
you may also need:
```
VisualChangesetModel.deleteMany({ experiment: experiment.id });
```

see: `deleteExperimentByIdForOrganization`

Reply:
Good catch. I've added this here be6ea4df